aspnet/publishing/linuxproduction.rst

```diff
@@ -0,0 +1,310 @@
+            tcp_nopush on;
+            tcp_nodelay off;
+            keepalive_timeout 65;
+            types_hash_max_size 2048;
+            # server_tokens off;
+
+            # server_names_hash_bucket_size 64;
+            # server_name_in_redirect off;
+
+            include /etc/nginx/mime.types;
+            default_type application/octet-stream;
+
+            ##
+            # Logging Settings
+            ##
+
+            access_log /var/log/nginx/access.log;
+            error_log /var/log/nginx/error.log;
+
+            ##
+            # Gzip Settings
+            ##
+
+            gzip on;
+            gzip_disable "msie6";
+
+            gzip_vary on;
+            gzip_proxied any;
+            gzip_comp_level 6;
+            gzip_buffers 16 8k;
+            gzip_http_version 1.0;
+            gzip_types text/plain text/css application/json application/x-javascript text/xml application/xml application/xml+rss text/javascript;
+
+            include /etc/nginx/conf.d/*.conf;
+            include /etc/nginx/sites-enabled/*;
+    }
+
+Once you have completed making changes to your nginx configuration you
+can run ``sudo nginx -t`` to verify the syntax of your configuration
+files. If you see any no errors you can ask nginx to pick up the changes
+by running ``sudo nginx -s reload``
+
+Create an user to run your application
+--------------------------------------
+
+For security reasons it is commonplace to run your application as an
+unprivileged user on Linux. Let us go ahead and create a unpriviliged
+user to run our ASP.NET web application
+
+.. code:: shell
+
+    sudo su
+    adduser --system --no-create-home --group aspnet
+    exit
+
+Install a DNX globally
+----------------------
+
+.. note::
+
+    This workaround is only required till `this change gets
+    merged <https://github.com/aspnet/dnvm/pull/393>`__
+
+.. code:: shell
+
+    # Grab a DNX from NuGet
+    dnvm upgrade
+
+    # Make it available globally
+    sudo mkdir -p /usr/local/lib/dnx/runtimes/
+    sudo cp -R ~/.dnx/runtimes/dnx-mono.1.0.0-beta6/ /usr/local/lib/dnx/runtimes/
+    sudo chmod 775 -R /usr/local/lib/dnx/runtimes/dnx-mono.1.0.0-beta6/
+    sudo chown -R aspnet /usr/local/lib/dnx/runtimes/dnx-mono.1.0.0-beta6/
+
+Start our web application on startup
+------------------------------------
+
+We will create an Upstart script to launch our application on startup.
+This will vary based on whether you are using System V Init, Upstart or
+Systemd as your startup system.
+
+Create a new startup script by running
+``sudo touch /etc/init/aspnet5webserver.conf``. Using your preferred
+text editor update the file contents to match what is shown below.
+
+.. code::
+
+    start on runlevel [2345]
+    stop on runlevel [016]
+
+    script
+        exec start-stop-daemon --start --make-pidfile --pidfile /var/run/aspnet5webserver.pid --chuid aspnet --exec /usr/local/lib/runtimes/dnx-mono.1.0.0-beta6/bin/dnx /path/to/packaged/app kestrel --signal INT
```

Comment:
`exec start-stop-daemon --start --make-pidfile --pidfile /var/run/aspnet5webserver.pid --chuid aspnet --exec /usr/local/lib/dnx/runtimes/dnx-coreclr-linux-x64.1.0.0-beta7/bin/dnx /path/to/packaged/app kestrel --signal INT`